ceexon/the-Questioner-Api
# #163243540 setup working area for api/v2

What this pr entails
---
1. Files and folders required for v2
2. Database schema setup and connection

Prerequisites
---
1. python3
2. flask
3. postgresql 

How to manually test
---
- clone this repo 
  `` https://github.com/kburudi/the-Questioner-Api/ ``
- install Postgres database
`` $ sudo apt-get install postgresql postgresql-contrib ``
- create a .env file
   `` $ touch .env ``
- create two Postgres databases in your virtual environment
      `` $ createdb <name_of_db> && createdb <name_of_test_db>
- add the following details to your .env file 
   ```
        export FLASK_APP="run.py"
        export SECRET="your_secret_key"
        export DB_URL="postgresql://<hostname>/<name_of_db>"
        export DB_HOST= <host_details> eg. (127.0.0.1)
        export DB_NAME="<name_of_db>"
        export DB_USER="<name_of_db_user>"
        export DB_PASSWORD="<db_user_password>"
  ```

- run the app
    `` $ flask run ``
""" YOU SHOULD RECEIVE A MESSAGE ON THE CONSOLE LIKE (" database creation complete ") """

Related PT stories 
---
**163243540**
https://www.pivotaltracker.com/n/projects/2235288
